week-3/implement/is-proper-fraction.test.js

```diff
@@ -0,0 +1,93 @@
+// You wil need to implement a function isProperFraction
+// You need to write assertions for your function to check it works in different cases
+
+// Terms:
+// Fractions: https://www.bbc.co.uk/bitesize/topics/zt9n6g8/articles/zjxpp4j
+// Written here like this: 1/2 == Numerator/Denominator
+
+// Acceptance criteria:
+
+// Proper Fraction check:
+// Input: numerator = 2, denominator = 3
+// target output: true
+// Explanation: The fraction 2/3 is a proper fraction, where the numerator is less than the denominator. The function should return true.
+
+// Improper Fraction check:
+// Input: numerator = 5, denominator = 2
+// target output: false
+// Explanation: The fraction 5/2 is an improper fraction, where the numerator is greater than or equal to the denominator. The function should return false.
+
+// Zero Denominator check:
+// Input: numerator = 3, denominator = 0
+// No target output: Error (Denominator cannot be zero)
+// Explanation: The function should throw an error when the denominator is zero, as it's not a valid fraction.
+
+// Negative Fraction check:
+// Input: numerator = -4, denominator = 7
+// target output: true
+// Explanation: The fraction -4/7 is a proper fraction because the absolute value of the numerator (4) is less than the denominator (7). The function should return true.
+
+// Equal Numerator and Denominator check:
+// Input: numerator = 3, denominator = 3
+// target output: false
+// Explanation: The fraction 3/3 is not a proper fraction because the numerator is equal to the denominator. The function should return false.
+
+// These acceptance criteria cover a range of scenarios to ensure that the isProperFraction function handles both proper and improper fractions correctly and handles potential errors such as a zero denominator.
+
+function isProperFraction(numerator, denominator) {
+  if (denominator === 0) {
+    return "Error: Denom cannot be 0";
+  } else if (numerator < denominator) {
+    return true;
+  } else if (numerator >= denominator) {
+    return false;
+  } else if (numerator < 0 && Math.abs(numerator) < denominator) {
+    return true;
+    //  condition checks if the fraction has a negative numerator. The Math.abs() function finds the absolute value, which is the positive size or amount of a number, no matter if it's negative or positive. So, for a negative number, its absolute value gives the positive size of that number.
+  } else if (numerator === denominator) {
+    return false;
+  }
```

Comment:
There are few points here:

1) you have few checks which return `true` and few which return `false`. Can you combine them to have only one check which returns `true` otherwise we just return `false`.

2) in your code you have:
```
} else if (numerator < denominator) {
...
} else if (numerator < 0 && Math.abs(numerator) < denominator) {
```
I'm almost sure that if you have have negative numerator (like -10) and positive denominator (like 2) then your first check returns `true` and you won't even reach the second check.
But I think you can avoid this problem if you re-organise the function as I suggested in 1)